### ISSUE TITLE
Failed to manage SR-IOV PF and VF ports, ignoring: Invalid argument

### ISSUE THREAD

**Issue by kroese** [systemd/systemd#27369]
### systemd version the issue has been seen with

252

### Used distribution

Ubuntu 23.04

### Linux kernel version used

5.19.0-40-generic

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

No errors

### Unexpected behaviour you saw

My log gets spammed by:

`systemd-networkd[554]: vethe8f1694: Failed to manage SR-IOV PF and VF ports, ignoring: Invalid argument`

I think the `vethe8f1694` interface is from Docker, but this issue started appearing when updating Ubuntu 22.10 to 23.04 so I suspect its more likely that its related to systemd than with docker.

### Steps to reproduce the problem

Honestly I have no idea. I have another system with exactly the same kernel/hardware/docker version, where I don't see this error, so I'm not sure what is causing this.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
systemd-networkd[554]: veth9634225: Failed to manage SR-IOV PF and VF ports, ignoring: Invalid argument
```

**Comment by hugalafutro** [comment]
For what it's worth I am seeing the same in logs occasionally when restarting / rebuilding stacks, and it too started after updating from ubuntu 22.10 to 23.04. This is inside a vm which has a pci device (igpu) passed through. No configuration other than what 
 do-release-upgrade did occured before first noticing the error in log. No functionality I utilize seems to be affected.

**Comment by hugalafutro** [comment]
@kroese Nope, the vm is used for docker containers of which it runs about 40, the only non-standard config in the underlying system I can think of (apart from it being a vm) is the whole machine is routed via wireguard tunnel (Mullvad vpn).

**Comment by yuwata** [comment]
Is the error shown in a docker container? Or in the host serving containers?
If it is found in the docker container, then that should be an issue in docker side, and I guess /sys is wrongly mounted in rw.

**Comment by hugalafutro** [comment]
@yuwata the issue was being reported on the host serving the containers, bringing containers up and down often triggered it for me, however not in a reproducible fashion (I have since migrated the docker containers to different host).

**Comment by yuwata** [comment]
1. Could you create the following config files?
- /etc/systemd/system/systemd-networkd.service.d/override.conf
```
[Service]
Environment=SYSTEMD_LOG_LEVEL=debug
```
- /etc/udev/rules.d/00-debug-net.rules
```
SUBSYSTEM=="net", OPTIONS="log_level=debug"
```
2. `systemctl restart systemd-networkd.service`, `systemctl reload systemd-udevd.service`, or reboot the machine.
3. then, please provide the log shown by `journalctl --no-hostname -u systemd-networkd.service -u systemd-udevd.service`.

**Comment by elafontaine** [comment]
There doesn't seems to be traction on this since May.  My understanding is that this isn't causing some issues per say, but that systemd-networkd is trying to manage the interface that were spawned from another process (e.g. docker).

I'm not too concerned about the traction, but I'm worried that this may lead to some side effects later down the line, should we try to remove the veth from the list of managed interfaces for networkd ? Is there a change coming that would do something to those interfaces ? (even though they're "unmanaged" (see bug on moby/moby) ).  

I would love to get some insights on this as I'm not a systemd-networkd guru.

**Comment by yuwata** [comment]
Ah, I reproduce the issue without docker, by using the following one-line script:
```sh
$ for ((i=0;i<100;i++)); do sudo ip link add veth99 type veth peer veth-peer; sudo ip link del veth99; done
```
Then, with #29116,
```
Sep 08 00:42:05 systemd-networkd[319943]: veth99: Failed to get 'dev_port' sysfs attribute: No such device
Sep 08 00:42:05 systemd-networkd[319943]: veth99: Failed to manage SR-IOV PF and VF ports, ignoring: No such device
Sep 08 00:42:05 systemd-networkd[319943]: veth99: udev initialized link
```

**Comment by yuwata** [comment]
Fix is waiting in #29116.

**Comment by yuwata** [comment]
(Note the warning is not critical, you can safely ignore them.)
